src/queue/multilevel.rs

```diff
@@ -266,57 +286,67 @@ impl LevelManager {
         extras.schedule_time = Some(now());
     }
 
-    fn get_level0_chance(&self) -> u32 {
-        self.level0_chance.load(SeqCst)
-    }
-
     fn maybe_adjust_chance(&self) {
         if self.adjusting.compare_and_swap(false, true, SeqCst) {
             return;
         }
         // The statistics may be not so accurate because we cannot load two
         // atomics at the same time.
-        let total = self.total_elapsed.0.load(SeqCst);
-        if Duration::from_micros(total) < ADJUST_CHANCE_INTERVAL {
-            // Another thread just adjusted the chances.
+        let total = self.total_elapsed_us.get();
```

Comment:
I think the order will be `Relaxed` for metrics, is it OK?

Reply:
I think it's OK as soon as the time of each handling is not very long so that the inaccuracy doesn't have much effect.

Reply:
I'm worry that `total_diff` can be zero as reorder is allowed now.

Reply:
If total_diff is zero, the function returns in L297. The calculation is different here and there is no risk of div by 0 like before. Anything else worrying?